src/test/java/org/mockito/internal/stubbing/answers/AbstractThrowsExceptionTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2020 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.stubbing.answers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+import static org.mockito.Mockito.mock;
+
+import java.io.IOException;
+import java.nio.charset.CharacterCodingException;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.mockito.exceptions.base.MockitoException;
+import org.mockito.internal.invocation.InvocationBuilder;
+import org.mockito.invocation.Invocation;
+
+public class AbstractThrowsExceptionTest {
+
+    @Test
+    public void should_raise_wanted_throwable() {
+        Throwable expected = new Exception();
+        AbstractThrowsException ate = instantiateFixture(expected);
+
+        Throwable throwable = Assertions.catchThrowable(() -> ate.answer(createMethodInvocation()));
+        assertNotNull("Should have raised an exception.", throwable);
+        assertSame(expected, throwable);
+    }
+
+    @Test
+    public void should_throw_mock_exception_without_stacktrace() {
+        AbstractThrowsException ate = instantiateFixture(mock(Exception.class));
+
+        Throwable throwable = Assertions.catchThrowable(() -> ate.answer(createMethodInvocation()));
+        assertNotNull("Should have raised an exception.", throwable);
+        assertThat(throwable.getStackTrace()).describedAs("no stack trace, it's mock").isNull();
+    }
+
+    @Test
+    public void should_fill_in_exception_stacktrace() {
+        AbstractThrowsException ate = instantiateFixture(new Exception());
+
+        Throwable throwable = Assertions.catchThrowable(() -> ate.answer(createMethodInvocation()));
+        assertNotNull("Should have raised an exception.", throwable);
+        assertThat(throwable.getStackTrace()[0].getClassName()).isEqualTo(AbstractThrowsException.class.getName());
+        assertThat(throwable.getStackTrace()[0].getMethodName()).isEqualTo("answer");
+    }
+
+    @Test(expected = MockitoException.class)
```

Comment:
Instead of `expected = `, use `assertThrows` and assert on the exception that it is of the correct exception class and it has the correct message.

Reply:
I've replaced `expected = ` but I can't find the import for `assertThrows`. I think the tests are configured for JUnit 4 only and `assertThrows` was introduced in 5. If you have Jupiter configured in another test source then I can migrate it to there and do the necessary alterations.

Reply:
Ah I misremembered the name. In AssertJ it is called `assertThatThrownBy`: https://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion